### ISSUE TITLE
Documentation suggestions: more on notebook creation and arrange to notebooks, instructor view, student view

### ISSUE THREAD

**Issue by rkdarst** [jupyter/nbgrader#1491]
I am at the point where I would like to write some better nbgrader docs  for my own team, but I thought perhaps I should contribute directly to nbgrader docs.  Since the changes are more significant than usual, I thought I would post an issue with the idea. (I would do most of these myself)

My fundamental goal is to make it easier to get someone started with nbgrader.  My main issue with the current documentation is that it doesn't seem to provide a gentle introduction to nbgrader for an outsider.  In particular, there aren't good examples of what a minimal notebook would look like (it is really simple, and that's the benefit!  It should be show as early as possible, and make the connection to unit testing).  Instead, most of the starting information is about the interface itself.

So, what I am thinking about right now is:
a) Interface highlights → add an example of what a notebook would look like.  Emphasize that this is like unit tests and the benefits of the single notebook format.
b) Also on "Interface highlights", divide it into three or four main topics: "creating notebooks", "managing assignments" (instructor view), "student view", and possibly "exchanging".
c) Add a "Creating notebooks" section before "Creating and grading assignments".  Move some of the material about notebooks to here.
d) Possibly rename/adjust the "User documentation" section to fit these three or four categories. I wouldn't do much, I think, but possibly try to clarify some of the names (eg. "Managing assignment files"→"Exchanging assignment files") or minor rearrangements.

Any thoughts on this, yes or no?  I doubt I would get all the way to the bottom of this list, but at least I can get started. If this isn't desired here, I would do at least a few of these in our own local (but open source) documentation.

**Comment by perllaghu** [comment]
Yes, definitely*...... I suspect there are several versions of this, in various levels of completeness & usefulness

We have ours at
* Instructor side: https://noteable.edina.ac.uk/nbguide/
* student side: https://noteable.edina.ac.uk/student-guide/

.... and there's some general stuff at https://noteable.edina.ac.uk/user-guide/

If there was a complete, central, repository for all this information, that would certainly save _me_ time in the future.

It would also be good to include some templates or examples to show how `### BEGIN SOLUTION` & `### END SOLUTION` work, what `### MARK SCHEME` does, what `### HIDDEN TESTS` do, and the difference between `manually-graded answer` and `manually-graded task` cells - again, there's some stuff to possibly use at https://github.com/edina/Exemplars2020/tree/master/TeachingDocs/

The thing I'm discovering more and more is that our service is not used by computing people nearly as much as by non-computing people - law & business studies & psychology. It's been..... "interesting"

[*] There's no such thing as too much documentation!

**Comment by rkdarst** [comment]
I agree with everything you say... especially the "needing to make it usable by less computer people".  I also have sometimes referred people to your documentation!  If we could make one place of high-quality documentation, would you want to combine yours with that?  Though, it seems like for your service, it would be best to create a dedicated user-docs site, separate from the developer documentation (it's too professional to combine into anything less, and I'm really wondering if we should even try to right now).

Similar to what you suggest with the examples, my major goal is to show something early that makes people realize how *cool* this is, by seeing the notebook with all the standard markup. and how it gets transformed.

**Comment by perllaghu** [comment]
Yes, absolutely..... 

I think all of the services that provide some form of centralised notebook service would like to have a central location so they don't need to duplicate pages and pages of documentation.

Being able to have some cool/clever examples of using existing notebook tricks & nbgrader tricks to made really good assignments will help. Nothing needs to be a big example, or particularly _wordy_ ..... just enough to show how something works, for a lecturer to take & expand on ..... kinda like `CodePen` or `jsFiddle` examples

We [the plural, each of us] will definitely need to have localised information that extends, or supersede this base set... how we handle multiple `course`s; if we change the basic path structure; if we change how/where autograding is done [I know Illumidesk are working on that]; etc.

..... but the idea that we, as a community, can produce useful documentation for teachers and students to make the best use of nbgrader.

In terms of _where_ - I see 3 options
1. keep in in nbgrader source, and on readthedocs
2. Create a new git repo that is purely for this documentation
3. Expand the documentation I already have

OK.... so perhaps number 3 is a bit tongue-in-cheek.....

**Comment by damianavila** [comment]
> Any thoughts on this, yes or no?

In general, 💯  about this idea.
